.travis.yml

```diff
@@ -1,18 +1,51 @@
-before_install:
-  - sudo apt-get update -myqq
-  - sudo apt-get install libcairo2-dev libjpeg8-dev libpango1.0-dev libgif-dev build-essential g++
 language: node_js
+sudo: false
+
 node_js:
-  - 0.10
+  - '0.10'
+  - '4'
+  - '5'
 
-branches:
-  only:
-    - master
+matrix:
+  allow_failures:
+    - node_js: '4'
+    - node_js: '5'
+
+cache:
+  directories:
+    node_modules
+
+after_script:
+  npm run coverage
 
 notifications:
   email:
     - thaddee.tyl@gmail.com
   irc: "irc.freenode.org#shields"
 
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      - g++-4.8
+      - libcairo2-dev
+      - libjpeg8-dev
+      - libpango1.0-dev
+      - libgif-dev
+      - build-essential
+
+before_install:
+  - npm -g install npm@latest-2
+  - npm explore npm -g -- npm install node-gyp@latest
```

Comment:
Travis new infrastructure is running on a container infrastructure.
Synce we need a C++ compiler, to build native module, we need to install it. (with the needed libraries for canvas)